nfc/libs/NfcCoreLib/lib/LibNfc/phLibNfc_Hci.c

```diff
@@ -1691,4 +1761,594 @@ static NFCSTATUS phHciNfc_CreateSEGetAtrTimer(phHciNfc_HciContext_t  *pHciContex
     }
     PH_LOG_LIBNFC_FUNC_EXIT();
     return wStatus;
+}
+
+/////////////////////////////////////////////////
+static NFCSTATUS phLibNfc_HciSetHostType(void* pContext, NFCSTATUS status, void* pInfo)
+{
+	NFCSTATUS wStatus = NFCSTATUS_INVALID_PARAMETER;
+	pphLibNfc_Context_t pLibCtx;
+	uint8_t bIndex = PHHCINFC_PIPE_HOST_TYPE_INDEX;
+	uint8_t aHostType[2] = { 0x01,0x00 };
+	UNUSED(pInfo);
+	UNUSED(status);
+	PH_LOG_LIBNFC_FUNC_ENTRY();
+	if (NULL != pContext)
+	{
+		pLibCtx = (pphLibNfc_Context_t)pContext;
+
+		if (NULL != pLibCtx->pHciContext)
+		{
+			wStatus = phHciNfc_AnySetParameter(pLibCtx->pHciContext,
+				bIndex,
+				PHHCINFC_PIPE_ADMIN_PIPEID,
+				2,
+				aHostType,
+				&phLibNfc_InternalSequence,
+				pContext);
+			if (wStatus != NFCSTATUS_PENDING)
+			{
+				PH_LOG_LIBNFC_CRIT_STR("phLibNfc_HciSetHostType:Failed to Send Cmd to NCI");
+			}
+			else
+			{
+				PH_LOG_LIBNFC_INFO_STR("phLibNfc_HciSetHostType:Cmd Sent to NCI");
+			}
+		}
+		else
+		{
+			PH_LOG_LIBNFC_CRIT_STR("phLibNfc_HciSetHostType: Invalid Hci context received!");
+		}
+	}
+	else
+	{
+		PH_LOG_LIBNFC_CRIT_STR("phLibNfc_HciSetHostType: Invalid LibNfc context received!");
+	}
+	PH_LOG_LIBNFC_FUNC_EXIT();
+	return wStatus;
+}
+
+static NFCSTATUS phLibNfc_HciSetHostTypeProc(void* pContext, NFCSTATUS status, void* pInfo)
+{
+	NFCSTATUS wStatus = NFCSTATUS_INVALID_PARAMETER;
+	pphLibNfc_Context_t pLibCtx;
+	UNUSED(pInfo);
+	PH_LOG_LIBNFC_FUNC_ENTRY();
+	if (NULL != pContext)
+	{
+		pLibCtx = (pphLibNfc_Context_t)pContext;
+		if (NULL != pLibCtx->pHciContext)
+		{
+			wStatus = status;
+			PH_LOG_LIBNFC_INFO_STR("phLibNfc_HciSetHostTypeProc: Sucessfull");
+			PH_LOG_LIBNFC_INFO_X32MSG("phLibNfc_HciSetHostTypeProc: Sucessfull ", wStatus);
+		}
+		else
+		{
+			PH_LOG_LIBNFC_CRIT_STR("phLibNfc_HciSetHostTypeProc: Invalid Hci context received!");
+		}
+	}
+	else
+	{
+		PH_LOG_LIBNFC_CRIT_STR("phLibNfc_HciSetHostTypeProc: Invalid LibNfc context received!");
+	}
+	PH_LOG_LIBNFC_FUNC_EXIT();
+	return wStatus;
+}
+
+static bool_t phHciNfc_Check_eSECompliancy(uint8_t *p_data, uint8_t data_len)
+{
+	uint8_t               noofhosts = 0;
+	uint8_t               count = 0;
+	uint8_t               host_id = 0;
+
+
+	noofhosts = ((data_len / PHHCINFC_PIPE_HOST_TYPE_LIST_LEN) - PHHCINFC_PIPE_HOST_TYPE_LIST_LEN);
+	PH_LOG_HCI_INFO_X32MSG("phHciNfc_Check_eSECompliancy: No of Hosts in the N/w:-", noofhosts);
+	for (count = 0; count < data_len; count++)
+	{
+		PH_LOG_HCI_INFO_X32MSG("phHciNfc_Check_eSECompliancy: Host Types :-", p_data[count]);
+	}
+	for (count = 0; count < noofhosts; count++)
+	{
+		host_id = (((count + 1) * PHHCINFC_PIPE_HOST_TYPE_LIST_LEN) + PHHCINFC_PIPE_HOST_TYPE_LIST_LEN);
+		PH_LOG_HCI_INFO_X32MSG("phHciNfc_Check_eSECompliancy: Host Type List Index :-", host_id);
+		if ((p_data[host_id] & p_data[host_id + 1]) != 0xFF)
+		{
+			if ((p_data[host_id] == 0x03) && (p_data[host_id + 1] == 0x00))
+			{
+				return TRUE;
+			}
+		}
+	}
```

Comment:
Correct me if I misunderstood this logic, but my current understanding of compliancy check code above:
```
if there is a host in the HOST_LIST with HOST_TYPE == '0300' (Embedded SE):
    proceed with ETSI12;
else:
    proceed with ETSI9;
```
Is that criteria by design?